Build failure: xl2tpd

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Build xl2ttpd package.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318227361

### Relevant log output

```console
build flags: SHELL=/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/bash PREFIX=\$\(out\)
gcc -DDEBUG_PPPD -DTRUST_PPPD_TO_DIE -Os -Wall -Wextra -DSANITY -DLINUX -I./linux/include/ -DUSE_KERNEL -DIP_ALLOCATION   -c -o xl2tpd.o xl2tpd.c
xl2tpd.c:91:35: error: initialization of 'int (*)(FILE *, char *)' from incompatible pointer type 'int (*)(void)' [-Wincompatible-pointer-types]
   91 |     {CONTROL_PIPE_REQ_LAC_STATUS, &control_handle_lac_status},
      |                                   ^
xl2tpd.c:91:35: note: (near initialization for 'control_handlers[10].handler')
xl2tpd.c:77:12: note: 'control_handle_lac_status' declared here
   77 | static int control_handle_lac_status();
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [<builtin>: xl2tpd.o] Error 1
```

### Additional context

- Related: https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.62-xanmod1, NixOS, 26.05 (Yarara), 26.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/8j9wnvk2ml85klhh7cad9szwhkzhfppm-nixos/nixos`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/xl/xl2tpd/package.nix
+++ b/pkgs/by-name/xl/xl2tpd/package.nix
@@ -25,6 +25,8 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "PREFIX=$(out)" ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types -std=gnu17";
+
   meta = {
     homepage = src.meta.homepage;
     description = "Layer 2 Tunnelling Protocol Daemon (RFC 2661)";

